utils.go

```diff
@@ -57,30 +40,29 @@ func JoinFromHeader(
 		return span, nil
 	}
 
-	// Else, fall back on SplitText.
-	carrier := NewSplitTextCarrier()
-	for key, val := range h {
-		if strings.HasPrefix(key, ContextIDHTTPHeaderPrefix) {
-			// We don't know what to do with anything beyond slice item v[0]:
-			unescaped, err := url.QueryUnescape(val[0])
-			if err != nil {
-				return nil, err
-			}
-			if carrier.TracerState == nil {
-				carrier.TracerState = map[string]string{}
-			}
-			carrier.TracerState[strings.TrimPrefix(key, ContextIDHTTPHeaderPrefix)] = unescaped
-		} else if strings.HasPrefix(key, TagsHTTPHeaderPrefix) {
-			// We don't know what to do with anything beyond slice item v[0]:
-			unescaped, err := url.QueryUnescape(val[0])
+	// Else, fall back on TextMap.
+	carrier := HTTPHeaderTextMapCarrier{h}
+	return tracer.Join(operationName, TextMap, carrier)
+}
+
+type HTTPHeaderTextMapCarrier struct {
```

Comment:
seems ok here, or utils. In theory we could also provide a wrapper for `map[string]string` to satisfy text map carrier.